ILogger_best_practice/kb/scenario_description/preamble.md

```diff
@@ -0,0 +1,49 @@
+# .NET ILogger logging with string interpolation using the $ symbol to compile-time logging source generation migration guide
+
+This article shows how to migrate .NET `ILogger` logging API calls from using `string interpolation using the $ symbol` to `compile-time logging source generation`.
+
+1.Identify the .NET `public interface ILogger` logging API calls that are using `string interpolation using the $ symbol` to do logging.
+
+Note: see the `String interpolation using the $ symbol` part to understand the patterns.
+
+2.For each identified call, migrate the code with following steps:
+
+- First: define partial methods for logging in your `Class`, using the LoggerMessage attribute. The attribute includes properties for `LogLevel`, and `Message`.
```

Comment:
It feels ambiguous what 'your Class' refers to. I assume it refers to one selected class that contained logging API calls identified in step (1), but a human or AI with no prior knowledge might not be able to make that connection?